tracer.go

```diff
@@ -45,20 +45,42 @@ const (
 )
 
 var (
-	// DefaultTracer is the default global Tracer, set at package
-	// initialization time, configured via environment variables.
-	//
-	// This will always be initialized to a non-nil value. If any
-	// of the environment variables are invalid, the corresponding
-	// errors will be logged to stderr and the default values will
-	// be used instead.
-	DefaultTracer *Tracer
+	tracerMu      sync.RWMutex
+	defaultTracer *Tracer
 )
 
-func init() {
+// DefaultTracer returns the default global Tracer, set the first time the
+// function is called. It is configured via environment variables.
+//
+// This will always be initialized to a non-nil value. If any of the
+// environment variables are invalid, the corresponding errors will be logged
+// to stderr and the default values will be used instead.
+func DefaultTracer() *Tracer {
+	tracerMu.RLock()
+	tracer := defaultTracer
+	tracerMu.RUnlock()
+	if tracer != nil {
+		return tracer
+	}
+
 	var opts TracerOptions
 	opts.initDefaults(true)
-	DefaultTracer = newTracer(opts)
+	tracer = newTracer(opts)
+	SetDefaultTracer(tracer)
+	return tracer
+}
+
+// SetDefaultTracer sets the tracer returned by DefaultTracer(). If another
+// tracer has already been initialized, any events are flushed and the tracer
```

Comment:
Comment needs updating now that we're not flushing. Probably best to explicitly state that queued events won't be flushed, and users need to call DefaultTracer().Flush() if they need to.